### PR TITLE
Feature/first version

### PR DESCRIPTION
**Pull request by jhedev**
Wrote a simple tool to automate lightsail snapshotting.

### CODE REVIEW

File: pkg/snapshot/snapshot.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+	"github.com/aws/aws-sdk-go/service/lightsail"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	defaultSnapshotSuffix = "auto-snapshot"
+)
+
+var (
+	defaultRetention = 10 * 25 * time.Hour
```
Comment (philippfranke) [inline pkg/snapshot/snapshot.go]: default retention is 10 days and 10 hours that's a weird duration 🤣. I'm fine with it. 😉

File: pkg/snapshot/snapshot.go [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+			continue
+		}
+		if snapshot.FromInstanceName != nil &&
+			*snapshot.FromInstanceName != smgr.instance {
+			// skip; wrong instance
+			continue
+		}
+
+		// Check if snapshot is old enough was created by this tool
+		if snapshot.CreatedAt.Before(time.Now().Add(-smgr.retention)) &&
```
Comment (philippfranke) [inline pkg/snapshot/snapshot.go]: Let's negate the first part of the if-statement and continue with the loop and maybe log that it's skipping a snapshot.

IMO, it's improving the readability.

File: pkg/snapshot/snapshot.go [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+			strings.HasSuffix(*snapshot.Name, smgr.suffix) {
```
Comment (philippfranke) [inline pkg/snapshot/snapshot.go]: Same here.

**Review by philippfranke — CHANGES_REQUESTED** [review]
Just a little change to improve the readability.

File: pkg/snapshot/snapshot.go (hunk 1 above)
Comment (jhedev) [reply pkg/snapshot/snapshot.go]: maybe 1337 hours would be better 🤔

File: pkg/snapshot/snapshot.go (hunk 2 above)
Comment (jhedev) [reply pkg/snapshot/snapshot.go]: Yeah, I thought about this as well after I had already submitted the PR. Will make some improvements here.

**Comment by jhedev** [comment]
Pushed some fixes @philippfranke. Please have a look again.

**Review by philippfranke — CHANGES_REQUESTED** [review]
LGTM so let's rebase your fixup.

**Comment by jhedev** [comment]
Done @philippfranke

**Review by philippfranke — APPROVED** [review]

**Pull request merged**